catalog/transformer/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/converter/mgmp/MgmpConverter.java

```diff
@@ -0,0 +1,872 @@
+     *
+     * @param metacard the source
+     * @return XstreamPathValueTracker containing XML paths and values to write
+     */
+    @Override
+    protected XstreamPathValueTracker buildPaths(MetacardImpl metacard) {
+
+        this.metacard = metacard;
+        pathValueTracker = new XstreamPathValueTracker();
+
+        addNamespaces();
+
+        addFileIdentifier();
+        addCharacterSet();
+        addHierarchyLevel();
+        addContact();
+        addDateStamp();
+        addMetadataStandardName();
+        addMetadataStandardVersion();
+        addCrs();
+        addMdIdentification();
+        addContentInfo();
+        addDistributionInfo();
+        addMetadataConstraints();
+
+        return pathValueTracker;
+    }
+
+    private void addMetadataConstraints() {
+        addMetadataSecurityClassification();
+        addOriginatorSecurity();
+        addMetadataReleasability();
+    }
+
+    private void addDistributionInfo() {
+        addGmdMetacardFormat();
+        addGmdResourceUri();
+    }
+
+    private void addContentInfo() {
+        addImageDescription();
+    }
+
+    private void addMdIdentification() {
+        addMdIdentificationCitation();
+        addMdIdentificationAbstract();
+        addMdIdentificationStatus();
+        addMdIdentificationPointOfContact();
+        addMdIdentificationDescriptiveKeywords();
+        addMdIdentificationResourceConstraints();
+        addMdIdentificationAggregationInfo();
+        addMdIdentificationLanguage();
+        addMdIdentificationTopicCategories();
+        addMdIdentificationExtent();
+    }
+
+    private void addMdIdentificationExtent() {
+        addGeographicIdentifier();
+        addBoundingPolygon();
+        addTemporalElements();
+        addVerticalElement();
+    }
+
+    private void addMdIdentificationResourceConstraints() {
+        addResourceSecurityClassification();
+        addMdIdentificationResourceConstraintsOriginatorClassification();
+        addMdIdentificationResourceConstraintsCaveats();
+    }
+
+    private void addMdIdentificationResourceConstraintsOriginatorClassification() {
+        addFieldIfString(Security.RESOURCE_ORIGINATOR_CLASSIFICATION,
+                MgmpConstants.RESOURCE_ORIGINATOR_SECURITY_PATH,
+                () -> {
+                    pathValueTracker.add(new Path(MgmpConstants.RESOURCE_ORIGINATOR_SECURITY_CODE_LIST_PATH),
+                            MgmpConstants.MGMP_CLASSIFICATION_CODE);
+                });
+    }
+
+    private void addMdIdentificationCitation() {
+        addGmdTitle();
+        addModifiedDate();
+        addCreatedDate();
+        addExpirationDate();
+    }
+
+    private void addMdIdentificationLanguage() {
+
+        List<String> languageCodes = getValues(Core.LANGUAGE);
+
+        if (languageCodes.isEmpty()) {
+            languageCodes = Collections.singletonList(Locale.ENGLISH.getISO3Language());
+        }
+
+        List<String> languageTexts = languageCodes.stream()
```

Comment:
changed to supply a descriptive language if possible, my tests indicate that if an unknown language code is sent (eg. "gere"), then the descriptive text will just be the language code